app/src/main/java/it/niedermann/owncloud/notes/persistence/migration/Migration_21_22.java

```diff
@@ -0,0 +1,46 @@
+package it.niedermann.owncloud.notes.persistence.migration;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.preference.PreferenceManager;
+import androidx.room.migration.Migration;
+import androidx.sqlite.db.SupportSQLiteDatabase;
+
+import it.niedermann.owncloud.notes.persistence.SyncWorker;
+
+/**
+ * Enabling backgroundSync, set from {@link String} values to {@link Boolean} values
+ * https://github.com/stefan-niedermann/nextcloud-notes/issues/1168
+ */
+
+public class Migration_21_22 extends Migration {
+    @NonNull
+    private final Context context;
+
+    public Migration_21_22(@NonNull Context context) {
+        super(21, 22);
+        this.context = context;
+    }
+
+    @Override
+    public void migrate(@NonNull SupportSQLiteDatabase database) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        if (sharedPreferences.contains("backgroundSync")) {
+            if (sharedPreferences.getString("backgroundSync", "on").equals("off")) {
+                editor.remove("backgroundSync");
+                editor.putBoolean("backgroundSync", false);
+                SyncWorker.update(context, false);
```

Comment:
No need to explicitly call it with `false` in this case, because if the background sync was explicitly switched to `off`, it is already disabled and the behavior will not change while migrating - just remove this line.